src/rotest/api/resource_control/query_resources.py

```diff
@@ -0,0 +1,63 @@
+# pylint: disable=unused-argument, no-self-use
+import httplib
+
+from django.db import transaction
+from django.db.models.query_utils import Q
+from swaggapi.api.builder.server.response import Response
+from swaggapi.api.builder.server.exceptions import BadRequest
+from swaggapi.api.builder.server.request import DjangoRequestView
+
+from rotest.management.common.json_parser import JSONParser
+from rotest.api.common.models import ResourceDescriptorModel
+from rotest.management.common.resource_descriptor import ResourceDescriptor
+from rotest.api.common.responses import (InfluencedResourcesResponseModel,
+                                         BadRequestResponseModel)
+
+
+class QueryResources(DjangoRequestView):
+    """Find and return the resources that answer the client's query.
+
+    Returns:
+        ResourcesReply. a reply containing matching resources.
+    """
+    URI = "resources/query_resources"
+    DEFAULT_MODEL = ResourceDescriptorModel
+    DEFAULT_RESPONSES = {
+        httplib.OK: InfluencedResourcesResponseModel,
+        httplib.BAD_REQUEST: BadRequestResponseModel
+    }
+    TAGS = {
+        "post": ["Resources"]
+    }
+
+    def post(self, request, *args, **kwargs):
+        """Find and return the resources that answer the client's query.
+
+        Args:
+            request (Request): QueryResources request.
+
+        Returns:
+            ResourcesReply. a reply containing matching resources.
+        """
+        try:
+            desc = ResourceDescriptor.decode(request.model.obj)
+
+        except Exception as e:
+            raise BadRequest(e.message)
```

Comment:
You might encounter a bug of ours, which should return 500 instead...

Reply:
What bug?